packages/server/api/src/app/app-connection/app-connection-utils.ts

```diff
@@ -1,11 +1,90 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BlockAuthProperty, PropertyType } from '@openops/blocks-framework';
 import {
   AppConnection,
+  AppConnectionType,
   AppConnectionWithoutSensitiveData,
+  RedactedAppConnection,
 } from '@openops/shared';
 
+const REDACTED_MESSAGE = '**REDACTED**';
+
 export const removeSensitiveData = (
   appConnection: AppConnection,
 ): AppConnectionWithoutSensitiveData => {
   const { value: _, ...appConnectionWithoutSensitiveData } = appConnection;
   return appConnectionWithoutSensitiveData as AppConnectionWithoutSensitiveData;
 };
+
+export function redactSecrets(
+  auth: BlockAuthProperty | undefined,
+  connection: AppConnection,
+): AppConnectionWithoutSensitiveData | RedactedAppConnection {
+  const value: Record<string, any> | undefined = connection.value;
+  if (!auth || !value) {
+    return removeSensitiveData(connection);
+  }
+
+  const redacted = { ...value };
+
+  switch (auth.type) {
+    case PropertyType.SECRET_TEXT: {
+      redacted.secret_text = REDACTED_MESSAGE;
+      return {
+        ...connection,
+        value: redacted as any,
+      };
+    }
+
+    case PropertyType.BASIC_AUTH: {
+      redacted.password = REDACTED_MESSAGE;
+      return {
+        ...connection,
+        value: redacted as any,
+      };
+    }
+
+    case PropertyType.CUSTOM_AUTH: {
+      if (redacted.props) {
+        const props = { ...redacted.props };
+        for (const [key, prop] of Object.entries(auth.props)) {
+          if (
+            (prop as { type: PropertyType }).type === PropertyType.SECRET_TEXT
+          ) {
+            props[key] = REDACTED_MESSAGE;
+          }
+        }
+        redacted.props = props;
+      }
+      return {
+        ...connection,
+        value: redacted as any,
+      };
+    }
+
+    case PropertyType.OAUTH2: {
+      if (
+        value?.type === AppConnectionType.OAUTH2 &&
+        typeof value.client_secret === 'string' &&
+        typeof value.client_id === 'string' &&
+        typeof value.redirect_url === 'string'
+      ) {
+        return {
+          ...connection,
+          value: {
+            type: PropertyType.OAUTH2,
+            client_id: value.client_id,
+            client_secret: REDACTED_MESSAGE,
+            redirect_url: value.redirect_url,
```

Comment:
so, for oauth 2 theres two options:

1) the user connects entierly through the pop up window
2) the user uses their own client id and password

Now, im not sure we need to return redirect_url, i know somewhere we take it from an env variable anyways but idk if the frontend or the backend does that